e2e/cypress/integration/api/iam/10_projects_api_action_update.ts

```diff
@@ -0,0 +1,157 @@
+
+  it('tagging 2 projects with all the attributes on a client runs action', () => {
+    const actionId = uuidv4();
+    const entityName = `entity_name-${Cypress.moment().format('MMDDYYhhmmss')}`;
+
+    // Ingest an action with attribues that match all the projects
+    cy.fixture('action/environment_create.json').then((action) => {
+      action.organization_name = '75th Rangers';
+      action.remote_hostname = 'example.org';
+      action.id = actionId;
+      action.entity_name = entityName;
+      action.recorded_at = Cypress.moment().utc().subtract(1, 'day').format();
+      cy.request({
+        headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+        method: 'POST',
+        url: '/data-collector/v0',
+        body: action
+      });
+    });
+
+    // wait for the action to be ingested
+    waitForAction(entityName, 30);
+
+    // create the projects with one rule
+    projectsWithRule.forEach(projectWithRule => {
+      cy.request({
+        headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+        method: 'POST',
+        url: '/apis/iam/v2beta/projects',
+        body: projectWithRule.project
+      });
+
+      cy.request({
+        headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+        method: 'POST',
+        url: `/apis/iam/v2beta/projects/${projectWithRule.rule.project_id}/rules`,
+        body: projectWithRule.rule
+      });
+    });
+
+    // apply rules
+    cy.request({
+      headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+      method: 'POST',
+      url: '/apis/iam/v2beta/apply-rules'
+    });
+
+    cy.waitUntilApplyRulesNotRunning(100);
+
+    const end = Cypress.moment().utc().endOf('day').valueOf().toString();
+    const start = Cypress.moment().utc().subtract(3, 'day').valueOf().toString();
+    // Ensure the event is tagged with the correct project
+    projectsWithRule.forEach(projectWithRule => {
+      cy.request({
+        headers: {
+          'api-token': Cypress.env('ADMIN_TOKEN'),
+          projects: projectWithRule.project.id
+        },
+        method: 'GET',
+        url: `api/v0/eventfeed?collapse=false&page_size=100&start=${start}&end=${end}`
+      }).then((response) => {
+        expect(response.body.events.length).to.greaterThan(0);
+        expect(eventExist(entityName, response.body.events)).to.equal(true);
+      });
+    });
+  });
+});
+
+function waitForAction(entityName: string, maxRetries: number) {
+  const end = Cypress.moment().utc().endOf('day').valueOf().toString();
+  const start = Cypress.moment().utc().subtract(3, 'day').valueOf().toString();
+
+  cy.request({
+    headers: {
+      projects: ['*'],
+      'api-token': Cypress.env('ADMIN_TOKEN')
+    },
+    method: 'GET',
+    url: `api/v0/eventfeed?collapse=false&page_size=100&start=${start}&end=${end}`
+  })
+  .then((resp: Cypress.ObjectLike) => {
+    // to avoid getting stuck in an infinite loop
+    if (maxRetries === 0) {
+      expect(0).to.equal(1);
```

Comment:
🙃 Maybe a comment? Not obvious to me what's happening.

Reply:
(hmm maybe this is old 🤔 )

Reply:
I think this may help, but I also suggest adding clarity to the existing comment.
```suggestion
      expect(maxRetries).to.not.equal(0);
```

Reply:
sorry forgot to leave a comment. these are from lance's PR I branched. it's merged now so I dropped the commit